packages/client/src/runtime/getPrismaClient.ts

```diff
@@ -1095,11 +1093,13 @@ new PrismaClient({
 
         if (globalThis.NOT_PRISMA_DATA_PROXY) {
           // async scope https://github.com/prisma/prisma/issues/3148
-          const resource = new AsyncResource('prisma-client-request')
-          return resource.runInAsyncScope(() => consumer(params))
+          return await new AsyncResource('prisma-client-request').runInAsyncScope(() => {
+            return runInChildSpan('request', tracer, internalParams.otelCtx, () => consumer(params))
```

Comment:
We have to manually pass the previously saved otel context because we can't propagate it automatically through prisma promises.

Reply:
Worth adding a comment in code?

Reply:
I added it closer to its context in https://github.com/prisma/prisma/pull/10764/files#diff-94c573b7cd12d73c919ef8300bc74e7536fb8cf26682c8409bf5cd8a9e6132f7R18